### PR TITLE
chore: improve setup

### PR DESCRIPTION
**Pull request by proquig**
## Description

This removes some common scripts and configurations which were duplicated between integrations
This adds `testcontainers` to run the docker compose stack before starting tests. As a consequence we don't need to configure it with environment variables and there won't be ports conflicts while running the stacks simultaneously.

## Additional Notes

N/A

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/slack/vitest.config.mts [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+
 export default defineConfig({
   test: {
-    setupFiles: ['./vitest/setup-database.ts', './vitest/setup-msw-handlers.ts'],
-    env: process.env,
-    // use 'node' if your integration is not compatible with edge runtime
-    environment: 'edge-runtime',
+    globalSetup: '@elba-security/test-utils/vitest/global-setup',
+    setupFiles: [
+      './vitest/setup-database.ts',
```
Comment (athielba) [inline apps/slack/vitest.config.mts:20]: I am wondering, why it is different from other saas setup (why not from packages) ?

File: packages/test-utils/src/vitest/global-setup.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const db = drizzle(sql);
+
+  await migrate(db, { migrationsFolder: './drizzle' }).finally(void sql.end);
+};
+
+export const teardown = async () => {
+  if (stack) {
+    await stack.down();
+  }
+};
```
Comment (athielba) [inline packages/test-utils/src/vitest/global-setup.ts:42]: Nice 👏🏼 👏🏼

File: drizzle.config.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import type { Config } from 'drizzle-kit';
+import { config } from 'dotenv';
+
+config({ path: '.env.local' });
+
```
Comment (athielba) [inline drizzle.config.ts:5]: if would be nice if you could check this here too
```
const { error } = config({ path: '.env.local' });

if (error) {
  throw new Error(`Could not find environment variables file: .env.local`);
}
```

**Review by athielba — APPROVED** [review]
LGTM, 👏🏼 , except few feedbacks

File: apps/slack/vitest.config.mts (hunk 1 above)
Comment (proquig) [reply apps/slack/vitest.config.mts:20]: As slack doesn't have a table name `organisations` but `teams` instead, I kept this script here to delete the data from this table before each test.

File: packages/test-utils/src/vitest/global-setup.ts (hunk 2 above)
Comment (elbalexandre) [reply packages/test-utils/src/vitest/global-setup.ts:42]: This is dope !

File: packages/test-utils/src/vitest/setup-database.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { drizzle } from 'drizzle-orm/postgres-js';
+import { sql } from 'drizzle-orm';
+
+// Delete every entries in the database between each tests
+beforeEach(async () => {
+  if (!process.env.DATABASE_URL) {
+    throw new Error('DATABASE_URL environment variable is not set');
+  }
+
+  const pg = postgres(process.env.DATABASE_URL);
```
Comment (elbalexandre) [inline packages/test-utils/src/vitest/setup-database.ts:12]: So the setup scripts are always running in node environment, good to know

**Review by elbalexandre — APPROVED** [review]

**Review by athielba — APPROVED** [review]

**Pull request merged**